### ISSUE TITLE
Channels affect cutoff frequency

### ISSUE THREAD

**Issue by airstruck** [camchenry/sone#1]
Something's wrong with the way the filter handles channels. I can make a mono and stereo sounddata that sound the same, but running them through the same filter produces different results. For example a 1-channel sound with a cutoff of 600 sounds the same as a 2-channel sound with a cutoff of 300.

I think sample rate needs to be multiplied by number of channels in the `w0` calculation.

**Comment by camchenry** [comment]
Do you have a .love that reproduces it? I think it might be because the filter sort of ignores the fact that a SoundData might be stereo.

**Comment by airstruck** [comment]
Here's a main.lua that shows it; try changing "channels" from 2 to 1, and changing the "frequency" parameter from 300 to 600. When changing the channel count, you'll also need to change the frequency to get the same sound. If you multiply `sr` by the number of channels in the `w0` calculation, you'll get the same sound for the same cutoff frequency regardless of number of channels.

``` lua
local sone = require 'sone'

local PI = math.pi
local INV_PI = 1 / PI
local TAU = 2 * PI

local seconds = 8
local rate = 44100
local bits = 16
local channels = 2
local samples = seconds * rate * channels

local sound = love.sound.newSoundData(samples, rate, bits, channels)

local timestep = 1 / rate
local angle = 0

local noteChange = math.floor(rate * channels * 0.25)
local freq = 100

for i = 0, samples * channels - channels, channels do
    if freq > 600 then freq = 100 end
    if i % noteChange == 0 then freq = freq * 1.5 end

    angle = (angle + freq * TAU * timestep) % TAU
    local square = angle > PI and 1 or -1 

    for j = 1, channels do
        sound:setSample(i + j - 1, square * 0.1)
    end
end

sone.filter(sound, { type = "lowpass", frequency = 300, Q = 5 })

local src = love.audio.newSource(sound)
src:play()

local visTop = 300
local visHeight = 30

function love.draw ()
    local pX, pY
    love.graphics.line(0, visTop - visHeight, 800, visTop - visHeight)
    love.graphics.line(0, visTop + visHeight, 800, visTop + visHeight)
    for x = 1, 800 do
        local y = sound:getSample(x * 8) * visHeight + visTop
        if pX then
            love.graphics.line(pX, pY, x, y)
        end
        pX, pY = x, y
    end
end
```

**Comment by airstruck** [comment]
I see what you mean about sone ignoring the channels though; multiplying sr by channels is just a band-aid, really what's needed is a separate filter instance for each channel, and each instance processes every nth sample.

**Comment by camchenry** [comment]
I think filter instances should fix this issue, I'll look into any optimizing it some point. Fixed by #2.
